misc/scripts/pre-commit

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+exec 1>&2
+exitVal=0
+while read -r f
+do
+  filename="${f##*/}"
+  extension="${filename##*.}"
+  p="$PWD/$f";
+  if [[ -f "$p" ]] && { [ "$extension" == "ql" ] || [ "$extension" == "qll" ]; }
```

Comment:
```suggestion
  if [[ -f "$p" ]] && { [ "$extension" == "ql" ] || [ "$extension" == "qll" ] }
```
What was this semicolon doing?

Reply:
It's the recommendation by Shellcheck. See https://github.com/koalaman/shellcheck/wiki/SC2235.

Reply:
Wow.